src/plugins/vis_type_timelion/public/timelion_vis_type.tsx

```diff
@@ -50,16 +51,19 @@ export function getTimelionVisDefinition(dependencies: TimelionVisDependencies)
     getSupportedTriggers: () => {
       return [VIS_EVENT_TO_TRIGGER.applyFilter];
     },
-    getUsedIndexPattern: (params: VisParams) => {
+    getUsedIndexPattern: async (params: VisParams) => {
       try {
-        const args = parseTimelionExpression(params.expression)?.args ?? [];
-        const indexArg = args.find(
-          ({ type, name, function: fn }) => type === 'namedArg' && fn === 'es' && name === 'index'
-        );
+        let indexPatterns: IndexPattern[] = [];
+        const indexArgs = getIndexArgs(params.expression);
 
-        if (indexArg?.value.text) {
-          return getIndexPatterns().find(indexArg.value.text);
+        for (const index in indexArgs) {
+          if (indexArgs[index]?.value.text) {
+            const foundIndexPatterns = await getIndexPatterns().find(indexArgs[index]?.value.text);
```

Comment:
extra `?`. you already checked it in `if (indexArgs[index]?.value.text) {`

Reply:
done